proposals/0000-config-finalization.md

```diff
@@ -0,0 +1,160 @@
+- Start Date: 2022-03-22
+- Reference Issues: [[Roadmap] Astro v1.0](https://github.com/withastro/rfcs/discussions/1)
+- Implementation PR: <!-- leave empty -->
+
+# Summary
+
+Astro's configuration file is something we'd like to clean up prior to a v1.0 Beta release. Currently, it is not evident which specific options should go in which area, leading to user confusion and ocassional duplication.
+
+This RFC proposes a clean-up of the existing Astro config file format for consistency and clarity.
+
+# Example
+
+When this RFC is accepted, a config file for Astro might look something like this:
+
+```js
+import { defineConfig } from 'astro';
+import preact from '@astrojs/preact';
+
+export default defineConfig({
+    log: 'info',
+
+    rootDir: '.',
+    srcDir: './src',
+    publicDir: './public',
+    outDir: './dist',
+
+    site: 'https://example.com',
+    trailingSlash: 'always',
+    basePath: '/',
+
+    build: {
+        format: 'file',
+    },
+
+    server: {
+        port: 3000
+    },
+
+    dev: {
+        host: '0.0.0.0',
+    },
+
+    preview: {
+        host: false,
+        port: 8080
+    },
+
+    markdown: {
+        drafts: true,
+        syntaxHighlight: 'shiki',
+    },
+
+    sitemap: {
```

Comment:
apologies if this is contradicting anything I said last week, but looking at this proposal now I think we should use this RFC to tackle the move of `sitemap` out of Astro core and into the `@astrojs/sitemap` integration. It feel silly to make a different breaking change in sitemap usage here if we know that move is coming anyway.

Reply:
I think this delineation makes it pretty clear that `markdown` and `sitemap` options are essentially integrations that come out-of-the-box. I think it's worth keeping both built-in for v1.0.

Reply:
Right, but I guess my question is "does sitemap need to be an integration that comes out-of-the-box?". Especially in a world where SSG vs. SSR is decided by your adapter and not always SSG.

Astro's `sitemap` feature is not currently powered by the integration, so there will be some extra implementation work on your end to stitch it all together if you go the "built-in" route.

Reply:
Updated to remove `sitemap` option. It should be configured by the `@astrojs/sitemap` integration.